### PR TITLE
Add Azure Application Credential

### LINKED ISSUE #22366: Create AzureApplicationCredential
This issue is to track the design and implementation of the `AzureImplementationCredential` for applications calling service APIs not suited for `DefaultAzureCredential`. Details can be found [here](https://gist.github.com/9793dbc036d16708fb7d8fe411fe0f1f)

### PR DESCRIPTION
**Pull request by g2vinay**
Fixes #22366

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureApplicationCredential.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * <ol>
+ * <li>{@link EnvironmentCredential}</li>
+ * <li>{@link ManagedIdentityCredential}</li>
+ * <li>Fails if none of the credentials above could be created.</li>
+ * </ol>
+ */
+@Immutable
+public final class AzureApplicationCredential extends ChainedTokenCredential {
+    /**
+     * Creates default AzureApplicationCredential instance to use. This will use environment variables to create
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureApplicationCredential.java:24]: remove default

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureApplicationCredentialBuilder.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.ArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+
+/**
+ * Fluent credential builder for instantiating a {@link AzureApplicationCredential}.
+ *
+ * @see AzureApplicationCredential
+ */
+public class AzureApplicationCredentialBuilder extends CredentialBuilderBase<AzureApplicationCredentialBuilder> {
```
Comment (srnagar) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureApplicationCredentialBuilder.java:18]: Make this `final`

**Review by srnagar — APPROVED** [review]

**Pull request merged**